### DISCUSSION TITLE
Allow energy sensors to report values in MJ

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#859] [Entity Models] [answered]
Proposal:
- Add `MJ` as a valid unit for energy sensors

Background:
- `MJ` is a valid energy unit, and is used in Australia for billing of gas. We already allow GJ.

Related core PRs:
- https://github.com/home-assistant/core/pull/86055
- https://github.com/home-assistant/core/pull/86061

**Comment by SgtBatten** [comment]
I'm here and keen to learn and discuss. 

I apologise for jumping the gun with the PR. I'd hoped it was a simple one given GJ was already implemented but understand that all things must be considered.

**Comment by frenck** [comment]
Which integrations would need this and use this today?

**Comment by jbouwh** [comment]
More and more often heat is delivered using a shared infrastructure that delivers heat though hot water and returning the cooler water. The energy issuing is measures in `GJ`. So I agree we should add Joules, but I think we will need `GJ` as well and may be `J`  too.

**Reply by frenck** [reply]
We already have GJ

**Reply by jbouwh** [reply]
I missed it here: https://developers.home-assistant.io/docs/core/entity/sensor/

**Reply by frenck** [reply]
Yeah, that has been missed. The PR creating the MJ updates it as well.

**Comment by emontnemery** [comment, marked as answer]
Let's add support for MJ as suggested in the description.
Adding support also for J can be handled separately.

**Comment by HAEdwin** [comment]
Sorry to break into this. From what I read here do I understand that **Heat Consumption** (City Heating) will or can become part of the Energy Dashboard just like Water and Gas? I convert the GJ readings to gas right now and this makes the tariff less accurate.

**Reply by SgtBatten** [reply]
I am billed for natural gas in MJ. 
What we've done is add MJ as a valid energy type just like GJ currently is. Not sure what city heating is sorry.

**Reply by HAEdwin** [reply]
City heating or District Heating is hot water being transported to houses where it is directly feeded into radiators or cold drinking water is passed through a heat converter for showering. Just like water and electricity this is a utility service where people are being billed in GJ. I do not get gas supplied to the home, cooking is done with electricity (induction).

**Reply by elupus** [reply]
@HAEdwin that would just be an Energy device class with GJ as unit. That is supported.

**Reply by HAEdwin** [reply]
@elupus I don't see how. In the Energy dashboard I can add Consumption (kWh), Solar Production (kWh), Battery System (kWh), Gas Consumption (ft3/m3), Water Consumption (m3/gl) or Individual Devices (kWh). There's no option for adding District Heating (gJ) or card named "Heat Consumption", as far as I can see. I'm using the MC66C integration (https://github.com/golles/Home-Assistant-Sensor-MC66C). Heat in Joules is not listed as device class https://www.home-assistant.io/integrations/sensor/#device-class
![image](https://user-images.githubusercontent.com/82239374/215090940-ab9902e5-ec96-4c02-8dda-023d72af89fe.png)

**Comment by elupus** [comment]
Ok. Right, yes only as individual entities now. This sounds like a separate discussion then.
